notebooks/working/6.0-mab-estimate-causal-effect.md

```diff
@@ -0,0 +1,2197 @@
+                                         alt_name_dict=alternative_name_dict)
+
+    wide_sim_data_availability[obs_id_column] = np.arange(wide_sim_data_availability.shape[0],
+                                            dtype=int) + 1
+    
+    print('Converting data from wide to long...')
+    long_sim_data = pl.convert_wide_to_long(wide_sim_data_availability,
+                                        ind_variables,
+                                        alt_varying_variables,
+                                        availability_variables,
+                                        obs_id_column,
+                                        choice_column,
+                                        new_alt_id_name=custom_alt_id)
+    
+    # Create a cars per licensed drivers column
+    long_sim_data["cars_per_licensed_drivers"] = 0
+    long_sim_data.loc[long_sim_data.num_licensed_drivers > 0,
+                      "cars_per_licensed_drivers"] = long_sim_data.num_cars / long_sim_data.num_licensed_drivers.astype(float)
+
+    # Add a variable representing cost divided by distance
+    long_sim_data["cost_per_distance"] = 0
+    long_sim_data.loc[long_sim_data.mode_id.isin([1, 2, 3]),
+                      "cost_per_distance"] = (long_sim_data.loc[long_sim_data.mode_id.isin([1, 2, 3]),
+                                                                "total_travel_cost"] /
+                                              long_sim_data.loc[long_sim_data.mode_id.isin([1, 2, 3]),
+                                                        "total_travel_distance"])
+    print('Simulating Choices...')
+    # Calculate probabilities for each alternative
+    # based on the estimated model
+    posterior_probs = mnl_model.predict(long_sim_data)
+    
+    
+    # Simulate choice data
+    long_sim_data['sim_choice'] = viz.simulate_choice_vector(posterior_probs,
+                               long_sim_data['observation_id'].values)
+    
+    
+    print('Estimating the choice model...')
+    # Estimate the basic MNL model, using the hessian and newton-conjugate gradient
+    mnl_model_sim = pl.create_choice_model(data=long_sim_data,
+                                           alt_id_col=alternative_id_col,
+                                           obs_id_col=observation_id_col,
+                                           choice_col=choice_column,
+                                           specification=mnl_specification,
+                                           model_type="MNL",
+                                           names=mnl_names)
+
+    num_vars = len(reduce(lambda x, y: x + y, mnl_names.values()))
+    # Note newton-cg used to ensure convergence to a point where gradient 
+    # is essentially zero for all dimensions. 
+    mnl_model_sim.fit_mle(np.zeros(num_vars),
+                          method="BFGS")
+    
+    mnl_model_sim.coefs.name = 'simulation_' + str(number)
+
+    models_dictionary[number] = mnl_model_sim
```

Comment:
So here you're still storing the model estimation results per simulation and then plotting those later.

We instead want to compute, for each simulation, the `naive_effect`, `estimated_effect`, and `true_effect`.
Again, we want to be sure to compute a scalar value for each of those on each simulation. You should then store these values in an array for later plotting.